### DISCUSSION TITLE
How much resources does an app use if it's in the background?

### DISCUSSION THREAD

**Discussion by d4h0** [emilk/egui#497] [Q&A] [answered]
Hi,

if I create a native app, does egui render the GUI even if the app isn't active?

For example, if I have an app that shows statistics, but I only check the app from time to time, will I still pay the full cost as if I'm interacting with the app, even if the app is in the background on a different workspace?

Best regards,
Daniel

**Comment by parasyte** [comment, marked as answer]
> if I create a native app, does egui render the GUI even if the app isn't active?

By default, no. `egui` widgets need to announce their intention to redraw by calling [`request_repaint()`](https://docs.rs/egui/0.12.0/egui/struct.Context.html#method.request_repaint) on the context reference. This is only necessary if the widgets are performing animations. Otherwise, `egui` only responds to input events like pointer movements and key presses.

**Reply by d4h0** [reply]
Thanks for answering my question, @parasyte!
